src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -2515,6 +2515,11 @@ public static NodeMigrationData MigrateShortestLacingToAutoLacing(NodeMigrationD
 
         #endregion
 
+        public virtual void OnBeforeAdd(Dynamo.Models.DynamoModel dynamoModel, WorkspaceModel workspace)
+        {
+            // Purposely empty. For inheritors to override if needed.
+        }
```

Comment:
In my mind it made sense to add this sort of life-cycle method to the node. It allowed to act at the right time and provides the required `DynamoModel`.